src/server.ts

```diff
@@ -81,3 +103,18 @@ newApp(
   .catch(err => {
     log.error("Error loading app: %s", err);
   });
+
+function appInsightsRequestListner(
+  app: Express,
+  telemetryClient?: appInsights.TelemetryClient
+): (_: http.IncomingMessage, __: http.ServerResponse) => void {
+  return (request: http.IncomingMessage, response: http.ServerResponse) => {
+    if (telemetryClient !== undefined) {
```

Comment:
this code is a little bit hard to grasp as the method called `appInsightsRequestListner` may return an express app _without_ a listener depending on the optional `telemetryClient` parameter. if it is possible I'd rather move this check up in the calling code. you may consider using an Option  type to avoid some undefined checks, something like:

```ts
const maybeAppinsightsClient = process.env.APPINSIGHTS_INSTRUMENTATIONKEY ? some(appInsightsClientBuilder.getClient())  : none();

....  .createServer(
          options,
          maybeAppinsightsClient .fold(
             _ => app,
             appInsightsClient => {
                appInsightsRequestListner(appInsightsClient, app)
             });
     )
....

server.on("close", () => {
      app.emit("server:stop");
      maybeAppinsightsClient .map(appInsightsClient => {
        appInsightsClient.flush();
        appInsights.dispose();
      });
      log.info("HTTP server close.");	      log.info("HTTP server close.");
});


```

Reply:
@gunzip fixed in commit 8f536bc